packages/react-components/react-tree/src/components/TreeItemAside/useTreeItemAside.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { getNativeElementProps, useMergedRefs } from '@fluentui/react-utilities';
+import type { TreeItemAsideProps, TreeItemAsideState } from './TreeItemAside.types';
+import { useTreeItemContext_unstable } from '../../contexts/treeItemContext';
+
+/**
+ * Create the state required to render TreeItemAside.
+ *
+ * The returned state can be modified with hooks such as useTreeItemAsideStyles_unstable,
+ * before being passed to renderTreeItemAside_unstable.
+ *
+ * @param props - props from this instance of TreeItemAside
+ * @param ref - reference to root HTMLElement of TreeItemAside
+ */
+export const useTreeItemAside_unstable = (
+  props: TreeItemAsideProps,
+  ref: React.Ref<HTMLElement>,
+): TreeItemAsideState => {
+  const isActionsVisible = useTreeItemContext_unstable(ctx => ctx.isActionsVisible);
+  const isAsideVisible = useTreeItemContext_unstable(ctx => ctx.isAsideVisible);
```

Comment:
nit: you can directly select the desired value from context

```suggestion
  const visible = useTreeItemContextUnstable(ctx => actions ? ctx.isActionsVisible : isAsideVisible)
```